packages/@aws-cdk/aws-servicediscovery/lib/instance.ts

```diff
@@ -0,0 +1,55 @@
+import cdk = require('@aws-cdk/cdk');
+import { IService } from './service';
+
+export interface IInstance extends cdk.IConstruct {
+  /**
+   * The id of the instance resource
+   */
+  readonly instanceId: string;
+
+  /**
+   * The Cloudmap service this resource is registered to.
+   */
+  readonly service: IService;
+}
+
+/**
+ * Used when the resource that's associated with the service instance is accessible using values other than an IP
+ * address or a domain name (CNAME), i.e. for non-ip-instances
+ */
+export interface BaseInstanceProps {
+  /**
+   * The id of the instance resource
+   */
+  instanceId: string;
+
+  /**
+   * Custom attributes of the instance.
+   *
+   * @default none
+   */
+  customAttributes?: object;
```

Comment:
Not sure if this translates well into JSII. It might, it might not.

Reply:
How do we handle arbitrary key-value pairs being passed in usually?

Reply:
If the values are strings, then `{[key: string]: string}` would do it.